lib/active_model/serializer/adapter/json_api/link.rb

```diff
@@ -3,30 +3,43 @@ class Serializer
     module Adapter
       class JsonApi
         class Link
-          def initialize(serializer)
+          def initialize(serializer, value)
             @object = serializer.object
             @scope = serializer.scope
+            if value.respond_to?(:call)
+              self._value = instance_eval(&value)
+            else
+              self._value = value
```

Comment:
The way I see it, providing a return value to the block should be interpreted as "I'm a big boy and this is my link value". Otherwise, "building" a link object through the DSL should be done, well, via the DSL. What are your thoughts?